providers/linux/arch_linux.go

```diff
@@ -19,9 +19,19 @@ package linux
 
 import (
 	"fmt"
+	"os"
+	"strings"
 	"syscall"
 )
 
+const (
+	procSysKernelArch = "/proc/sys/kernel/arch"
```

Comment:
Any idea when this was introduced? Empirically it seems this is relatively new (spot checking a 5.19.17 host and it wasn't there).

Reply:
this one was tricky to find, I've played substituting the version in the url https://www.kernel.org/doc/html/v6.1/admin-guide/sysctl/kernel.html#arch

the lowest number which douments it is 6.1

Reply:
So, the docs say: `The machine hardware name, the same output as uname -m (e.g. x86_64 or aarch64).` If both `Architecture` and `NativeArchitecture` are always going to be the same, we should either leave one unimplemented or document this more clearly.

Reply:
I can't change Kernel documentation. I'm relaying here on the fact that although behind `/proc` filesystem are the same system calls made, only the actual syscalls are emulated, whilst `/proc` behaves like a regular filesystem for the process running in emulation. We will have `Architecture` and `NativeArchitecture`, which is self explanatory. In case we can't retrieve the native architecture, we'll leave it blank.

Reply:
added more inline doc 542d3ca573ba8252f0d3709ceda2a80f522dd00a